🐛 fix: correct the model list on the model provider's page

lc-soft opened the pull request:
#### 💻 变更类型 | Change Type

<!-- For change type, change [ ] to [x]. -->

- [ ] ✨ feat
- [x] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 👷 build
- [ ] ⚡️ perf
- [ ] 📝 docs
- [ ] 🔨 chore

#### 🔀 变更说明 | Description of Change

模型列表去重时是以 `/` 后的名称作为唯一键的：
When deduplicating model lists, the name after the `/` is used as the unique key:

![image](https://github.com/user-attachments/assets/373495ca-bf74-4670-97d4-2a3f81064f35)

以 Qwen/Qwen2-7B-Instruct 为例，Gitee AI 中记录的模型 id 是 `Qwen2-7B-Instruct`，其它提供商的模型 id 是 `Qwen/Qwen2-7B-Instruct`，这种情况下，靠判断 `item.identifier === id` 是找不到 Gitee AI 的 `Qwen2-7B-Instruct` 模型详细的。

Using Qwen/Qwen2-7B-Instruct as an example, the model ID recorded in Gitee AI is `Qwen2-7B-Instruct`, while the model ID for other providers is `Qwen/Qwen2-7B-Instruct`. In this case, it is not possible to find detailed information about the Gitee AI's `Qwen2-7B-Instruct` model by determining that `item.identifier === id`.

因此，改成与 `getModelList()` 中相同的方式获取 id 再进行判断。
Therefore, it is changed to obtain the id in the same way as in `getModelList()` and then make a judgment.

#### 📝 补充信息 | Additional Information

**Before:**

The total number of models is 4 **+2**

![model-list-before](https://github.com/user-attachments/assets/c3021d9a-964d-49f2-880f-e54682fd0d0a)

**After:**

The total number of models is 4 **+10**

![model-list-after](https://github.com/user-attachments/assets/f725446e-e787-4d62-8c93-9d616fbac881)

lc-soft committed:
fix: correct the model list on the model provider's page

canisminor1990 commented:
@lc-soft 影响到了其他 Provider ，尝试改成这样试试？

如：https://lobe-chat-preview-git-fork-lc-soft-l-50c801-lobe-chat-community.vercel.app/discover/provider/openrouter

```ts
let model = list.find((item) => {
      const ids = item.identifier.split('/');
      return ids[1] === id  || item.identifier === id;
});
```

lc-soft commented:
> @lc-soft 影响到了其他 Provider ，尝试改成这样试试？
> 
> 如：https://lobe-chat-preview-git-fork-lc-soft-l-50c801-lobe-chat-community.vercel.app/discover/provider/openrouter
> 
> ```ts
> let model = list.find((item) => {
>       const ids = item.identifier.split('/');
>       return ids[1] === id  || item.identifier === id;
> });
> ```

改了。

我测试时看到有些 id 包含多个斜杆 `/`，这种 id 的模型可能仍然存在找不到的情况。

![image](https://github.com/user-attachments/assets/bbd2b345-d576-4ed1-b2a1-7e0e952a030d)

canisminor1990 commented:
那改成include呢

lc-soft commented:
> 那改成include呢

修改 `getModelList()` 里的 `uniqBy()`，怎样？

```js
return uniqBy(list, (item) => item.identifier.split('/').pop());
```

查找用 includes()：

```js
const name = id.split('/').pop();
let model = list.find((item) => item.identifier.includes(name));
```

例如：

```js
[
  { identifier: '@cf/qwen/qwen1.5-14-b-chat-awq' },
  { identifier: 'qwen1.5-14-b-chat-awq' }
]
```

去重后：

```js
[
  { identifier: '@cf/qwen/qwen1.5-14-b-chat-awq' },
]
```

查找结果：

```text
@cf/qwen/qwen1.5-14-b-chat-awq -> { identifier: '@cf/qwen/qwen1.5-14-b-chat-awq' }
qwen1.5-14-b-chat-awq -> { identifier: '@cf/qwen/qwen1.5-14-b-chat-awq' }
```
